File: scripts/process_logo.py
from PIL import Image
import os

src = r"C:\Users\PKS\.cursor\projects\c-Users-PKS-source-repos-SalonPro-Backend\assets\c__Users_PKS_AppData_Roaming_Cursor_User_workspaceStorage_6449116ebc088c0d665701169c39c3ac_images_imagesss-53a7c72c-5873-4d90-bb57-199f9da44774.png"
out_dir = r"c:\Users\PKS\source\repos\ElsInt\web\public\images\brand"
out_api = r"c:\Users\PKS\source\repos\ElsInt\src\ElsInt.Api\wwwroot\images\brand"
os.makedirs(out_dir, exist_ok=True)
os.makedirs(out_api, exist_ok=True)

img = Image.open(src).convert("RGBA")
pixels = img.load()
w, h = img.size

corners = [pixels[2, 2], pixels[w - 3, 2], pixels[2, h - 3], pixels[w - 3, h - 3]]
br = sum(c[0] for c in corners) // 4
bg = sum(c[1] for c in corners) // 4
bb = sum(c[2] for c in corners) // 4


def dist(p):
    return abs(p[0] - br) + abs(p[1] - bg) + abs(p[2] - bb)


for y in range(h):
    for x in range(w):
        r, g, b, a = pixels[x, y]
        d = dist((r, g, b))
        if d < 50:
            pixels[x, y] = (r, g, b, 0)
        elif d < 85:
            alpha = int(255 * (d - 50) / 35)
            pixels[x, y] = (r, g, b, max(0, min(255, alpha)))

bbox = img.getbbox()
if bbox:
    pad = 16
    l, t, r, b = bbox
    l = max(0, l - pad)
    t = max(0, t - pad)
    r = min(w, r + pad)
    b = min(h, b + pad)
    img = img.crop((l, t, r, b))

# Slightly taller header-friendly max height via resize keeping aspect
max_h = 320
if img.height > max_h:
    ratio = max_h / img.height
    img = img.resize((int(img.width * ratio), max_h), Image.Resampling.LANCZOS)

path1 = os.path.join(out_dir, "elsint-logo.png")
path2 = os.path.join(out_api, "elsint-logo.png")
img.save(path1)
img.save(path2)
print("saved", path1, img.size)

File: scripts/process-logo.mjs
import { Jimp, intToRGBA } from 'jimp';
import path from 'path';
import fs from 'fs';

const src = String.raw`C:\Users\PKS\.cursor\projects\c-Users-PKS-source-repos-SalonPro-Backend\assets\c__Users_PKS_AppData_Roaming_Cursor_User_workspaceStorage_6449116ebc088c0d665701169c39c3ac_images_imagesss-53a7c72c-5873-4d90-bb57-199f9da44774.png`;
const outs = [
  String.raw`c:\Users\PKS\source\repos\ElsInt\web\public\images\brand\elsint-logo.png`,
  String.raw`c:\Users\PKS\source\repos\ElsInt\src\ElsInt.Api\wwwroot\images\brand\elsint-logo.png`,
];

const img = await Jimp.read(src);
const { r: br, g: bg, b: bb } = intToRGBA(img.getPixelColor(4, 4));

img.scan(0, 0, img.bitmap.width, img.bitmap.height, function (x, y, idx) {
  const r = this.bitmap.data[idx];
  const g = this.bitmap.data[idx + 1];
  const b = this.bitmap.data[idx + 2];
  const d = Math.abs(r - br) + Math.abs(g - bg) + Math.abs(b - bb);
  if (d < 55) {
    this.bitmap.data[idx + 3] = 0;
  } else if (d < 95) {
    this.bitmap.data[idx + 3] = Math.round(255 * ((d - 55) / 40));
  }
});

let minX = img.bitmap.width,
  minY = img.bitmap.height,
  maxX = 0,
  maxY = 0;
img.scan(0, 0, img.bitmap.width, img.bitmap.height, function (x, y, idx) {
  if (this.bitmap.data[idx + 3] > 20) {
    if (x < minX) minX = x;
    if (y < minY) minY = y;
    if (x > maxX) maxX = x;
    if (y > maxY) maxY = y;
  }
});

const pad = 18;
minX = Math.max(0, minX - pad);
minY = Math.max(0, minY - pad);
maxX = Math.min(img.bitmap.width - 1, maxX + pad);
maxY = Math.min(img.bitmap.height - 1, maxY + pad);
img.crop({ x: minX, y: minY, w: maxX - minX + 1, h: maxY - minY + 1 });

const targetH = 280;
const scale = targetH / img.bitmap.height;
img.resize({ w: Math.round(img.bitmap.width * scale), h: targetH });

for (const out of outs) {
  fs.mkdirSync(path.dirname(out), { recursive: true });
  await img.write(out);
}
console.log('done', img.bitmap.width, img.bitmap.height, 'bg', br, bg, bb);
